weaviate/collections/batch/batch_wrapper.py

```diff
@@ -23,6 +30,8 @@ def __init__(self, connection: ConnectionV4, consistency_level: Optional[Consist
 
         self._batch_data = _BatchDataWrapper()
 
+        self._event_loop = event_loop
```

Comment:
why are we not using the singleton here, like everywhere else?

Reply:
I felt that it was cleaner to have the singleton called only once here and injected where it was needed. In the case of the decorator, that's the only place the singleton is called then the decorator is added wherever. In this case, I'd like the same logic in that we call the singleton once and then apply it everywhere it needs to go

Reply:
I'm not 100% sure if I can follow 😅 would you mind expanding a bit?

Reply:
I've revisited this code and you're right, it's more confusing prop drilling here and not in other places. Have changed in the latest version!